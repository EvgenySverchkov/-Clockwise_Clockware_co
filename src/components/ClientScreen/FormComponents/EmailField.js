import React from "react";

const EmailField = ({id}) => (
    <div className="col-sm-8 col-md-8">
        <input id={id} type="email" name="email" className="form-control" required />
    </div>
);

export default EmailField;