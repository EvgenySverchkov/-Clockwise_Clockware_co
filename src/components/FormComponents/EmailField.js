import React from "react";
import PropTypes from "prop-types";

const EmailField = ({ id, chngHandler, value }) => (
  <div className="col-sm-8 col-md-8">
    <input
      id={id}
      type="email"
      name="email"
      className="form-control"
      onChange={chngHandler}
      value={value}
      required
    />
  </div>
);

EmailField.propTypes = {
  id: PropTypes.string,
  chngHandler: PropTypes.func,
  value: PropTypes.string,
};

export default EmailField;
