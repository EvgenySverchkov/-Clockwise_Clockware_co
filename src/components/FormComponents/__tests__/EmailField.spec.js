import React from 'react';
import {create} from "react-test-renderer";

import EmailField from "../EmailField";
import ContextComponent from "../../../ContextComponent";

it("<EmailField /> test which is used in forms", ()=>{
    const mockProps = {
        id: "1",
        chngHandler: jest.fn(), 
        value: "some@mail.com", 
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <EmailField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
});