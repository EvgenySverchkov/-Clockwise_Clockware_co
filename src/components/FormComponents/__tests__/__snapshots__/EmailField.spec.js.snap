// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailField /> test which is used in forms 1`] = `
<div
  className="col-sm-8 col-md-8"
>
  <input
    aria-describedby={null}
    className="form-control"
    id="1"
    name="email"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    readOnly={false}
    required={true}
    title="Message"
    type="email"
    value="some@mail.com"
  />
</div>
`;
